presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java

```diff
@@ -102,7 +102,7 @@ public ReorderJoins(CostComparator costComparator)
         }
 
         MultiJoinNode multiJoinNode = toMultiJoinNode(joinNode, lookup);
-        JoinEnumerationResult result = new JoinEnumerator(idAllocator, symbolAllocator, session, lookup, multiJoinNode.getFilter(), costComparator).chooseJoinOrder(multiJoinNode.getSources(), multiJoinNode.getOutputSymbols());
```

Comment:
I think this approach is good too, but I thought it will be simpler initially to just check if enumerated plan has cross join and if so, then return original plan